Sources/Cache/Cache.swift

```diff
@@ -0,0 +1,159 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseAuth
+
+// FDC field name in server response that identifies a GlobalID
+let GlobalIDKey: String = "_id"
+
+// Client cache that internally uses a CacheProvider to store content.
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+actor Cache {
+  let config: CacheSettings
+  let dataConnect: DataConnect
+
+  private var cacheProvider: CacheProvider?
+
+  // holding it to avoid dealloc
+  private var authChangeListenerProtocol: NSObjectProtocol?
```

Comment:
cc: @peterfriese, no action required–– just highlighting how to currently integrate with the auth listener